### PR TITLE
ISO 2022-jp encoding and decoding tests

### PR DESCRIPTION
**Pull request by domenic**
I was inspired to do another round of shepherding @r12a's tests through the process. This is an updated version of #3199 with fixes to make lint pass and JavaScript prettification.

It reportedly passes in Firefox 57, per https://github.com/w3c/web-platform-tests/pull/3199#issuecomment-322538629, so probably @hsivonen is happy with them at least, and we should merge them. Maybe @hsivonen or @inexorabletash can approve this so that, once CI passes, I can merge them.

Closes #3199.

### CODE REVIEW

File: encoding/legacy-mb-japanese/iso-2022-jp/iso2022jp-encode-form-csiso2022jp.html.headers [hunk 1]
```diff
@@ -0,0 +1 @@
+Content-Type: text/html; charset=csiso2022jp
```
Comment (inexorabletash) [inline encoding/legacy-mb-japanese/iso-2022-jp/iso2022jp-encode-form-csiso2022jp.html.headers]: Can you add a trailing newline here and the other .headers files?

It looks like that was done for the shift_jis tests, and makes `cat`ing the files more pleasant

File: encoding/legacy-mb-japanese/iso-2022-jp/iso2022jp-encode-href-ascii.html [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            out += String.fromCodePoint(
+                parseInt(str.charAt(ptr + 1) + str.charAt(ptr + 2), 16)
+            );
+            ptr += 2;
+        } else out += str.charAt(ptr);
+        ptr++;
+    }
+}
+
+function encode(input, expected, desc) {
```
Comment (inexorabletash) [inline encoding/legacy-mb-japanese/iso-2022-jp/iso2022jp-encode-href-ascii.html]: Used in 5 test files; would be nice to factor it out but we can do that later.

**Review by inexorabletash — APPROVED** [review]
Let's get 'em in!

File: encoding/legacy-mb-japanese/iso-2022-jp/iso2022jp-encode-href-ascii.html (hunk 2 above)
Comment (domenic) [reply encoding/legacy-mb-japanese/iso-2022-jp/iso2022jp-encode-href-ascii.html]: Yeah I think there's a general project to be done where you diff files in these encoding tests against each other and factor out large chunks, or use the .sub.html infrastructure to just make them into one test and then iframe N copies of them.

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/20188)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/297785449)

**Pull request merged**